cache/subtopic.go

```diff
@@ -0,0 +1,55 @@
+package cache
+
+import (
+	"strings"
+	"sync"
+)
+
+// SubtopicsIDs contains a list of subtopics in map form with addition to mutex locking
+// The subtopicsMap is used to keep a record of subtopics to be later used to generate the subtopics id `query` for a topic
+// and to check if the subtopic id given by a user exists
+type SubtopicsIDs struct {
+	sync.Mutex
+	subtopicsMap map[string]bool
+}
+
+// GetNewSubTopicsMap creates a new subtopics id map to store subtopic ids with addition to mutex locking
+func NewSubTopicsMap() *SubtopicsIDs {
+	return &SubtopicsIDs{
+		subtopicsMap: make(map[string]bool),
+	}
+}
+
+// Get returns a bool value for the given key (id) to inform that the subtopic id exists
+func (t *SubtopicsIDs) Get(key string) bool {
+	t.Lock()
+	defer t.Unlock()
+
```

Comment:
(minor): I think you can allow multiple reads here as it is thread safe, so you could use a ```RWMutex``` in stead of ```Mutex``` here. With ```R.Lock() and R.unlock``` ?